script.js

```diff
@@ -1,52 +1,68 @@
 const formBtn = document.querySelector('#form-button');
 const booksList = document.querySelector('.books-list');
+let books = JSON.parse(localStorage.getItem('books'));
 
-let books = [];
+class Book {
+  constructor(id, title, author) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+  }
 
-const booksAdd = (id, title, author) => {
-  const li = document.createElement('li');
-  li.innerHTML = `<h2>${title}</h2>
-    <h3>${author}</h3>
-    <hr>`;
-  const deleteBtn = document.createElement('button');
-  deleteBtn.textContent = 'Remove';
-  li.insertBefore(deleteBtn, li.lastElementChild);
-  booksList.appendChild(li);
-  deleteBtn.addEventListener('click', () => {
-    books = books.filter((books) => {
-      if (books.title !== title && books.author !== author) {
+  addBook() {
+    const { title, author } = this;
+    const bookObj = { title, author };
+    if (title !== '' || author !== '') {
+      books.push(bookObj);
```

Comment:
- [ ] Your app happens to break after refactoring your code. An error is thrown in an attempt to add new book details. This appears to be due to anonymous abject used in your code. Kindly ensure to improve upon this aspect.

**Error whiles adding object:**
![image](https://user-images.githubusercontent.com/12745474/211767963-cfb28c52-fa11-46ed-9cd1-d4be0127b22f.png)

Reply:
Duly noted and changes would be implemented. Thanks for the review